src/MPI.jl

```diff
@@ -160,8 +161,15 @@ using PrecompileTools: @compile_workload
     @compile_workload begin
         # Running the load time hooks here shaves off a significant amount of loading time,
         # see also https://github.com/JuliaParallel/MPI.jl/pull/728
-        foreach(hook -> hook(), _mpi_load_time_hooks)
+        foreach(call, _mpi_load_time_hooks)
     end
+    # We insert some explicit precompile statements here. The corresponding methods
+    # are likely to be called from anyone using MPI in Julia. Thus, it is reasonable
+    # to compile them ahead of time instead of compiling them in parallel for each
+    # run.
+    Base.precompile(Tuple{typeof(Init)})
+    Base.precompile(Tuple{typeof(Comm_size), Comm})
+    Base.precompile(Tuple{typeof(Comm_size), Comm})
```

Comment:
```suggestion
    Base.precompile(Init, ())
    Base.precompile(Comm_size, (Comm,))
    Base.precompile(Comm_size, (Comm,))
```
Change as suggested by @giordano